torch/_inductor/codegen/triton.py

```diff
@@ -1327,19 +1327,8 @@ def final_argreduce(buffer, result_var, value, index):
                 )
                 final_argreduce(self.suffix, result_var, accumulator, accumulator_index)
             else:
-                if reduction_type == "min":
-                    updated = f"triton_helpers.minimum({accumulator}, {value})"
-                elif reduction_type == "max":
-                    updated = f"triton_helpers.maximum({accumulator}, {value})"
-                elif reduction_type == "sum":
-                    updated = f"{accumulator} + {value}"
-                elif reduction_type == "prod":
-                    updated = f"{accumulator} * {value}"
-                elif reduction_type == "xor_sum":
-                    updated = f"{accumulator} ^ {value}"
-                else:
-                    raise NotImplementedError(f"reduction_type {reduction_type}")
-
+                combine_fn = ir.get_reduction_combine_fn(reduction_type, src_dtype)
+                updated = combine_fn(accumulator, value)
                 self.compute.writeline(
                     f"{accumulator} = tl.where({cond}, {updated}, {accumulator})"
                 )
```

Comment:
Idea, but not for this PR. If we could know that `value` has just been loaded, we could elide this `tl.where`, as in those cases, the `other=` kwarg within `tl.load` already sets those extra values to the correct identity element.

I don't think we have a nice way of doing this now though, as the CSE vars carry close to no information.